File: Greedy/1829-maximum-units-on-a-truck/maximum-units-on-a-truck.py
class Solution:
    def maximumUnits(self, boxTypes: List[List[int]], truckSize: int) -> int:
        boxTypes.sort(key=lambda x: x[1], reverse=True)
        
        totalUnits = 0
        size = 0

        for box in boxTypes:
            if size + box[0] > truckSize:
                totalUnits += box[1] * (truckSize - size)
                break
            else:
                totalUnits += box[0] * box[1]
                size += box[0]
        
        return totalUnits

File: Greedy/1829-maximum-units-on-a-truck/maximum-units-on-a-truck.cpp
class Solution {
public:
    int maximumUnits(vector<vector<int>>& boxTypes, int truckSize) {
        sort(boxTypes.begin(), boxTypes.end(), [](vector<int>& a, vector<int>& b){
            return a[1] > b[1];
        });

        int totalUnits = 0, size = 0, i = 0;

        for(int i = 0; i < boxTypes.size() && size < truckSize; i++){
            if(size + boxTypes[i][0] > truckSize){
                totalUnits += boxTypes[i][1] * (truckSize - size);
                break; 
            }
            else{
                totalUnits += boxTypes[i][0] * boxTypes[i][1];
                size += boxTypes[i][0];
            }
        }

        return totalUnits;
    }
};